runtime/runtime-params-estimator/src/vm_estimator.rs

```diff
@@ -130,6 +135,128 @@ pub fn load_and_compile(
+    let workdir = tempfile::Builder::new().prefix("runtime_testbed").tempdir().unwrap();
+    let store = create_store(&get_store_path(workdir.path()));
+    let cache_store = Arc::new(StoreCompiledContractCache { store });
+    let cache: Option<&dyn CompiledContractCache> = Some(cache_store.as_ref());
+
+    let mut xs = vec![];
+    let mut ys = vec![];
+
+    // We use core-contracts, e2f60b5b0930a9df2c413e1460e179c65c8876e3.
+    // File 341191, code 279965, data 56627.
+    let data = include_bytes!("../test-contract/res/lockup_contract.wasm");
+    let contract = ContractCode::new(data.to_vec(), None);
+    xs.push(data.len() as u64);
+    ys.push(measure_contract(vm_kind, gas_metric, &contract, cache));
+
+    // File 257516, code 203545, data 50419.
+    let data = include_bytes!("../test-contract/res/staking_pool.wasm");
+    let contract = ContractCode::new(data.to_vec(), None);
+    xs.push(data.len() as u64);
+    ys.push(measure_contract(vm_kind, gas_metric, &contract, cache));
+
+    // File 135358, code 113152, data 19520.
+    let data = include_bytes!("../test-contract/res/voting_contract.wasm");
+    let contract = ContractCode::new(data.to_vec(), None);
+    xs.push(data.len() as u64);
+    ys.push(measure_contract(vm_kind, gas_metric, &contract, cache));
+
+    // File 124250, code 103473, data 18176.
+    let data = include_bytes!("../test-contract/res/whitelist.wasm");
+    let contract = ContractCode::new(data.to_vec(), None);
+    xs.push(data.len() as u64);
+    ys.push(measure_contract(vm_kind, gas_metric, &contract, cache));
+
+    // Least squares method.
+    let n = xs.len();
+    let n128 = n as i128;
+
+    let mut sum_prod = 0 as i128; // Sum of x * y.
+    for i in 0..n {
+        sum_prod = sum_prod + (xs[i] as i128) * (ys[i] as i128);
+    }
+
+    let mut sum_x = 0 as i128; // Sum of x.
+    for i in 0..n {
+        sum_x = sum_x + (xs[i] as i128);
+    }
+
+    let mut sum_y = 0 as i128; // Sum of y.
+    for i in 0..n {
+        sum_y = sum_y + (ys[i] as i128);
+    }
+
+    let mut sum_x_square = 0 as i128; // Sum of x^2.
+    for i in 0..n {
+        sum_x_square = sum_x_square + (xs[i] as i128) * (xs[i] as i128);
+    }
+
+    let b = Ratio::new(n128 * sum_prod - sum_x * sum_y, n128 * sum_x_square - sum_x * sum_x);
+    let a = Ratio::new(sum_y * b.denom() - b.numer() * sum_x, n128 * b.denom());
+
+    // Compute error estimation.
+    let mut error = 0i128;
+    for i in 0..n {
+        let expect = (a + b * (xs[i] as i128)).to_integer();
+        let diff = expect - (ys[i] as i128);
+        error = error + diff * diff;
+    }
+    println!("Error {}", (error as f64).sqrt() / (n as f64));
+
+    // We multiply `b` by 5/4 to accommodate for the fact that test contracts are typically 80% code,
+    // so in the worst case it could grow to 100% and our costs are still properly estimate.
+    // (a, b * Ratio::new(5i128, 4i128))
```

Comment:
I think above math formula is correct, however, it doesn't make sense to do a lsm on varied portion of code section / size.
Assume you test with 
x1 = file size of file1.wasm, which has 90% of code section, y1 = cost
x2 = file size of file2.wasm, which has 60% of code section, y2 = cost
...
Then what is does the `y = mx + b` means? it would give both errorneous estimation on both 90%-code and 60% code contract, and the est doesn't give you an upper bound. So i suggest do two set of least square method:
x1 = code size of file1.wasm, y = cost1
x2 = code size of file2.wasm, y = cost2
...
You get a cost = m1 * code_size + b1
Then do least square on: xi = code size of file_i.wasm, y = cost_i, and get cost = m2 * data_size + b2.
Then, cost = max(m1,m2)*file_size + max(b1,b2) should give an upper bound cost, no matter how many portion of the file is code or data